core/src/execution/datafusion/expressions/utils.rs

```diff
@@ -81,12 +81,12 @@ pub fn array_with_timezone(
     array: ArrayRef,
     timezone: String,
     to_type: Option<&DataType>,
-) -> ArrayRef {
+) -> Result<ArrayRef, ArrowError> {
```

Comment:
Should we return an Err instead of panicking in the not supported case?

Reply:
Could you expand on which case you are referring to here? Because I thought making it return an error instead of panicking is exactly what I did in the PR.